### PR TITLE
Implement `_lfortran_read_array_double`

### LINKED ISSUE #2119: code generation error: Real arrays of kind 4 only supported for now. Found kind: 8
```
subroutine abc
      integer :: m, n, k
      double precision :: a(m, n, k)
      read(10) a
end subroutine
program name
end program name
```

```console
(lf) harshita@harshita-linux:~/Desktop/lfortran$ lfortran a.f90
code generation error: Real arrays of kind 4 only supported for now. Found kind: 8
```

What does this mean? Does it mean that the read function is implemented for arrays of kind 4 and not for 8?
In that case, if I do
```
subroutine abc
      integer :: m, n, k
      integer :: a(m, n, k)
      read(10) a
end subroutine
program name
end program name
```
I get a seg fault as following:
```console
(lf) harshita@harshita-linux:~/Desktop/lfortran$ lfortran a.f90
lfortran: /home/harshita/miniconda3/envs/lf/include/llvm/IR/Instructions.h:897: llvm::Type* llvm::checkGEPType(llvm::Type*): Assertion `Ty && "Invalid GetElementPtrInst indices for type!"' failed.
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #2119.

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -1028,3 +1028,5 @@ RUN(NAME cpp_pre_02 LABELS gfortran llvm c wasm CPP_PRE)
 RUN(NAME dabs_01 LABELS gfortran llvmImplicit)
 
 RUN(NAME minpack_03 LABELS gfortran llvm NOFAST)
+
+RUN(NAME read_array_01 LABELS gfortran llvm NOFAST COPY_TO_BIN read_array_01.txt)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:1032]: I am slightly concerned about the new naming sequence. Please, could you reuse the existing naming sequence:

https://github.com/lfortran/lfortran/blob/a2d36b14b8492a2fd188915e53906ad34e862dc8/integration_tests/CMakeLists.txt#L801-L807

File: integration_tests/read_array_01.f90 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    open(unit=j, file="read_array_01.txt")
+    read(j, *) a
+    close(j)
+    if (abs(a(1, 1, 1) - 1.0D0) > 1e-10) error stop
+    if (abs(a(1, 2, 1) - 2.0D0) > 1e-10) error stop
+    if (abs(a(1, 1, 2) - 3.0D0) > 1e-10) error stop
+    if (abs(a(1, 2, 2) - 4.0D0) > 1e-10) error stop
+    if (abs(a(1, 1, 3) - 5.0D0) > 1e-10) error stop
+    if (abs(a(1, 2, 3) - 6.0D0) > 1e-10) error stop
+    print *, a
```
Comment (Shaikh-Ubaid) [inline integration_tests/read_array_01.f90:14]: Please use the print before the `if() error stop`. This helps during debugging as there would be a value printed on the console before the assertion failure.

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (Pranavchiku) [reply integration_tests/CMakeLists.txt:1032]: Ah yes, thank you for pointing this out. I was searching for a naming sequence, but I didn't find that at the moment. I will do the changes.

File: integration_tests/CMakeLists.txt [hunk 3]
```diff
@@ -1019,3 +1019,5 @@ RUN(NAME cpp_pre_02 LABELS gfortran llvm c wasm
 RUN(NAME dabs_01 LABELS gfortran llvmImplicit)
 
 RUN(NAME minpack_03 LABELS gfortran llvm NOFAST)
+
+RUN(NAME file_08 LABELS gfortran llvm NOFAST COPY_TO_BIN file_08.txt)
```
Comment (Shaikh-Ubaid) [inline integration_tests/CMakeLists.txt:1023]: @Pranavchiku I think this should be placed along with the existing `file_*` test cases (below the existing `file_07` test case on line `807` in `integration_tests/CMakeLists.txt`).

Comment (Shaikh-Ubaid) [reply integration_tests/CMakeLists.txt:1023]: And probably, we can rename `file_08.txt` to `file_08_data.txt` just like we have the existing files `file_01_data.txt` and `file_03_data.txt`.

https://github.com/lfortran/lfortran/blob/a2d36b14b8492a2fd188915e53906ad34e862dc8/integration_tests/CMakeLists.txt#L801-L803

I think following a constant naming culture helps improve maintainability over the longer term.

**Review by Shaikh-Ubaid — APPROVED** [review]
I shared a concern about test case placement and file renaming here https://github.com/lfortran/lfortran/pull/2155#discussion_r1292725349. Apart from it, this looks good to me. Thank you so much @Pranavchiku for fixing this.

**Review by certik — APPROVED** [review]
I think that this is good. Tests can be renamed / rearranged in subsequent PRs.

**Pull request merged**